### PR TITLE
Basic Use Case tests for the Presentation API

### PR DESCRIPTION
**Pull request by mariuswsl**
Tests for the Use Cases 'monitor availability of presentation displays', 'starting a new presentation' and 'reconnect to a presentation' for the Presentation API.

### CODE REVIEW

**Comment by zqzhang** [comment]
w3c-test:mirror

File: presentation-api/controlling-ua/reconnectToPresentation_success_manualTest.html [hunk 1]
```diff
@@ -0,0 +1,78 @@
+<!DOCTYPE html>
```
Comment (zqzhang) [inline presentation-api/controlling-ua/reconnectToPresentation_success_manualTest.html]: s/_manualTest/-manual/ in file name. See

http://testthewebforward.org/docs/manual-test.html

File: presentation-api/controlling-ua/reconnectToPresentation_success_manualTest.html [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            if (localStorage["presId"]) var presId = localStorage["presId"];
+
+            return request.reconnect(presId)
+                    .then(function (connection) {
+
+                        test(function () {
+                            assert_true(connection instanceof PresentationConnection);
+                        }, 'The connection is an instance of PresentationConnection.');
+
+                        assert_true(true);
```
Comment (zqzhang) [inline presentation-api/controlling-ua/reconnectToPresentation_success_manualTest.html]: No need this assertion.

File: presentation-api/controlling-ua/reconnectToPresentation_success_manualTest.html [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        promise_test(function () {
+
+            // get the saved presentation id
+            // for the reconnect function if exist
+            if (localStorage["presId"]) var presId = localStorage["presId"];
+
+            return request.reconnect(presId)
+                    .then(function (connection) {
+
+                        test(function () {
```
Comment (zqzhang) [inline presentation-api/controlling-ua/reconnectToPresentation_success_manualTest.html]: Actually no need to make a new test here.

File: presentation-api/controlling-ua/startNewPresentation_error-manuelTest.html [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+            request = new PresentationRequest(validPresURL);
+
+    // -------------------------------------------
+    // Start New Presentation Test (error) - begin
+    // -------------------------------------------
+    var startPresentation = function () {
+        promise_test(function () {
+            return request.start()
+                    .catch(function (err) {
+                        if (typeof SecurityError == "undefined" || typeof OperationError == "undefined") {
```
Comment (zqzhang) [inline presentation-api/controlling-ua/startNewPresentation_error-manuelTest.html]: Seems these two errors are not defined in global. You will always get true here, I think.

File: presentation-api/controlling-ua/startNewPresentation_error.html [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+            validPresURL = 'https://google.com/cast#__castAppId__=CC1AD845/__castClientId__=' + validUnixDate,
+            request = new PresentationRequest(validPresURL);
+
+    // -----------------------------------
+    // Start New Presentation Test - begin
+    // -----------------------------------
+    promise_test(function () {
+        return request.start()
+                .catch(function (error) {
+                    if (typeof InvalidAccessError == "undefined") {
```
Comment (zqzhang) [inline presentation-api/controlling-ua/startNewPresentation_error.html]: Use an assertion instead of the `if ... else`.

**Comment by louaybassbouss** [comment]
@mariuswsl  in the Presentation API F2F Meeting we discussed also the way how we create Presentation URLs that are compatible with Google Cast Receivers and other Receiving UAs. We need to replace in all tests `'https://google.com/cast#__castAppId__=CC1AD845/__castClientId__=' + client_id;` with `'http://w3c-test.org/presentation-api/receiving-ua/idlharness.html#__castAppId__=2334D33A/__castClientId__=' + client_id;`. the Cast Id for `http://w3c-test.org/presentation-api/receiving-ua/idlharness.html` is in this case `2334D33A`. This will work for now in Chrome canary https://www.google.com/chrome/browser/canary.html. Other Implementations like crosswalk from Intel will still work with this solution and the hash parameters will be ignored.

**Comment by zqzhang** [comment]
> 'http://w3c-test.org/presentation-api/receiving-ua/idlharness.html#__castAppId__=2334D33A/__castClientId__=' + client_id;

Can we use relative presentation URL?

**Comment by louaybassbouss** [comment]
@zqzhang agree relative URL is better if we want to run the tests from a local system and not from w3c-tests.org. @mariuswsl  please consider this comment.

File: presentation-api/controlling-ua/startNewPresentation_error-manual.html [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+            request = new PresentationRequest(validPresURL);
+
+    // -------------------------------------------
+    // Start New Presentation Test (error) - begin
+    // -------------------------------------------
+    var startPresentation = function () {
+        promise_test(function () {
+            return request.start()
+                    .catch(function () {
+                        assert_true(true);
```
Comment (zqzhang) [inline presentation-api/controlling-ua/startNewPresentation_error-manual.html]: Better check something here?

File: presentation-api/controlling-ua/startNewPresentation_error-manual.html [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    // -------------------------------------------
+    // Start New Presentation Test (error) - begin
+    // -------------------------------------------
+    var startPresentation = function () {
+        promise_test(function () {
+            return request.start()
+                    .catch(function (err) {
+
+                        test(function () {
+                            assert_true(err.code == 20, 'The error code is equal to 20.');
```
Comment (zqzhang) [inline presentation-api/controlling-ua/startNewPresentation_error-manual.html]: Better avoid magic number and use assert_equals.

You can also remove these 2 test wrapers; just make these 3 assertions in the promise test.

**Comment by zqzhang** [comment]
Merge this as-is so that you can regenerate a testing report and address the comment in following pull requests.

**Pull request merged**